native/src/main/java/io/ballerina/stdlib/java.jms/producer/Actions.java

```diff
@@ -113,23 +116,25 @@ public static Object sendTo(Environment env, BObject producer, BObject session,
                                 Message message) {
         MessageProducer nativeProducer = (MessageProducer) producer.getNativeData(NATIVE_PRODUCER);
         Session nativeSession = (Session) session.getNativeData(NATIVE_SESSION);
-        Future balFuture = env.markAsync();
-        executorService.execute(() -> {
-            try {
-                Destination jmsDestination = getDestination(nativeSession, destination);
-                nativeProducer.send(jmsDestination, message);
-                balFuture.complete(null);
-            } catch (BallerinaJmsException exception) {
-                BError bError = createError(JMS_ERROR, exception.getMessage(), exception);
-                balFuture.complete(bError);
-            } catch (UnsupportedOperationException | JMSException exception) {
-                BError bError = createError(JMS_ERROR,
-                        String.format("Error occurred while sending a message to the JMS provider: %s",
-                                exception.getMessage()), exception);
-                balFuture.complete(bError);
-            }
+        return env.yieldAndRun(() -> {
+            CompletableFuture<Object> balFuture = new CompletableFuture<>();
+            executorService.execute(() -> {
```

Comment:
Can we use virtual thread here?

Reply:
Hope this works 3b97839f45f16295e5c3d962f2cce4153cf92174